models/models_file.py

```diff
@@ -0,0 +1,168 @@
+            self.method = Method(method.upper())
+        except ValueError as e:
+            logger.error("Your config file passed invalid method: %s", e)
+            raise SystemExit
+
+    def __repr__(self) -> str:
+        return f"Response {self.method} after {self.latency} {self.status_code}: {self.body}"
+
+    @classmethod
+    def fromDict(cls, data: dict, method: str) -> object:
+        body = data["body"]
+        statusCode = data["statusCode"]
+        method = method
+        latency = data["latency"]
+        return cls(statusCode, body, method, latency)
+
+
+class Endpoint:
+    def __init__(self, endpoint: str, responses: "List[Response]") -> None:
+        self.responses = responses
+        self.endpoint = "/" + endpoint
+
+    def __repr__(self) -> str:
+        return f"Endpoint: \"{self.endpoint}\" responses: {self.responses}"
+
+    @classmethod
+    def fromDict(cls, data: dict) -> object:
+        endpoint = cls.adapt_route(data['endpoint'])
+        response = data["responses"][0]
+        responses = [Response.fromDict(response, data["method"])]
+        return cls(endpoint, responses)
+
+    @classmethod
+    def responseFromDict(cls, data: dict) -> object:
+        response = data["responses"][0]
+        return Response.fromDict(response, data["method"])
+
+    @classmethod
+    def adapt_route(cls, route: str) -> str:
+        name_allowed = "A-z0-9_"
+        params_search = re.compile(f"(:[{name_allowed}]+)")
+        params = params_search.findall(route)
+        for param in params:
+            route = route.replace(param, f"<{param[1:]}>", 1)
+        return route
+
+    def view_maker(self) -> object:
+        def view_func(*args, **kwargs):
+            for response in self.responses:
+                if request.method == response.method.value:
+                    sleep(response.latency / 1000)
+                    body = response.body
+                    param_dict = {"urlParam": kwargs, "queryParam": request.args}
+                    params = ReSearching.search_params(body)
+                    for param in params:
+                        total = param[0]
+                        type = param[1]
+                        var_name = param[2].strip()[1:-1]
+                        default = param[3][1:-1]
+                        replacement = (
+                            param_dict[type][var_name] if var_name in param_dict[type] else default)
+                        body = body.replace(total, replacement, 1)
+                    kwargs_dict = {"method": request.method, "hostname": request.host, "ip": request.remote_addr}
+                    kwparams = ReSearching.search_keywoards(body)
+                    for kwparam in kwparams:
+                        total = kwparam[0]
+                        name = kwparam[1]
+                        if name in kwargs_dict: body = body.replace(total, kwargs_dict[name], 1)
+                    return jsonify(eval(body)), response.status_code.value
```

Comment:
Uuuuuu, dlaczego tu musi iść eval? Nie boisz się, że ktoś dostarczy jakiś kod, którego na serwerze byśmy nie chcieli uruchomić?

Reply:
Ten eval jest tutaj, gdyż mockoon przekazuje body jako string bez usuniętych żadnych znaków specjalnych. Aby zwrócić mimetype "application/json" używam funkcji jsonify, w której "For security reasons only objects are supported toplevel. For more information about this, have a look at JSON Security.". A ten eval i tak bierze tylko pod uwagę część pliku konfiguracyjnego mockoona który jest dostarczany przez osoby które tak czy siak mają dostęp do kodu.